docs/docs/sourcing-from-wordpress.md

```diff
@@ -2,7 +2,119 @@
 
-This is a stub. Help our community expand it.
+This guide will walk you trough the process of using [Gatsby](https://www.gatsbyjs.org/) with [Wordpress Rest Api](https://developer.wordpress.org/rest-api/reference/).
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+WordPress is a free and open-source content management system (CMS). Let's say you have a site built with Wordpress and you want to pull the existing data into your static Gatsby site. You can do that with [gatsby-source-wordpress](https://www.gatsbyjs.org/packages/gatsby-source-wordpress/?=wordpress). Let's begin!
+
+_Note: this guide uses the `gatsby-starter-default` to provide you with a knowledge necessary to start working with Wordpress but if you get stuck at some point of the guide feel free to use
+[this example](https://github.com/gatsbyjs/gatsby/tree/master/examples/using-wordpress) to gain extra insights._
+
+## Setup
+
+###Bootstrapping a project
+
+First, you need to make sure that you have installed Gatsby CLI. To do that run the following command in shell:
+
+```shell
+npm install -g gatsby-cli
+```
+
+Now you are ready to create a default starter project that you will use to work with Wordpress and proceed to newly created directory:
+
+```shell
+gatsby new gatsby-wordpress-example
+cd gatsby-wordpress-example
+```
+
+That gets you to a new site directory with a couple (mostly) empty “gatsby”-prefixed files and a src directory with some basic scaffolding. The configuration and lifecycle hooks for Gatsby get put in those “gatsby”-prefixed files, you will need to have a closer look at `gatsby-config.js.`.
+
+###gatsby-config.js
+
+Essentially the Gatsby home base. The two things defined here initially (in the starter) are `siteMetadata` and `plugins` you can add to enable new functionalities on your site.
+
+```javascript
+module.exports = {
+  siteMetadata: {
+    title: "Gatsby Default Starter",
+  },
+  plugins: ["gatsby-plugin-react-helmet"],
+  ...
+}
+```
+
+###Plugin: gatsby-source-wordpress
+
+Now that you have some understanding of project structure lets add fetching Wordpress data functionality. There's a plugin for that. [`gatsby-source-wordpress`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-wordpress) is Gatsby's plugin for sourcing data from WordPress sites using the WordPress JSON REST API. You can install it by running the following command:
+
+```shell
+npm install gatsby-source-wordpress
+```
+
+###Configuring plugin
+
+In `gatsby-config.js`, add your configuration options, including your WordPress site's baseUrl, protocol, whether it's hosted on [wordpress.com](http://wordpress.com/) or self-hosted, and whether it makes use of the Advanced Custom Fields (ACF) plugin.
+
+```javascript
+module.exports = {
+  ...
+  plugins: [
+    ...,
+    {
+      resolve: `gatsby-source-wordpress`,
+      options: {
+        // your wordpress source
+        baseUrl: `amberley.me`,
```

Comment:
i know this is just pulled from the blog post, but let's make this something random (wpexample.com)? 👍